src/content/blog/astro-570.mdx

```diff
@@ -0,0 +1,305 @@
+
+This experimental feature allows you to use fonts from both your file system and several built-in supported providers (e.g. Google, Fontsource, Bunny) through a unified API. Keep your site performant thanks to sensible defaults and automatic optimizations including fallback font generation.
+
+To enable this feature, configure an `experimental.fonts` object with one or more fonts:
+
+```js title="astro.config.mjs"
+import { defineConfig, fontProviders } from "astro/config"
+
+export default defineConfig({
+    experimental: {
+        fonts: [{
+            provider: fontProviders.google(),
+            name: "Roboto",
+            cssVariable: "--font-roboto",
+        }]
+    }
+})
+```
+
+Then, add a `<Font />` component and site-wide styling in your `<head>`:
+
+```astro title="src/components/Head.astro"
+---
+import { Font } from 'astro:assets'
+---
+<Font cssVariable='--font-roboto' preload />
+<style>
+body {
+    font-family: var(--font-roboto);
+}
+</style>
+```
+
+There's so much more to discover! Head over to [the experimental Fonts documentation](https://docs.astro.build/en/reference/experimental-flags/fonts/) for the full API, how to get started, and even how to build your own custom `AstroFontProvider` if we don't yet support your preferred font service.
+
+We'd also appreciate your feedback. Participate in the ongoing development of this feature on the [Fonts RFC](https://github.com/withastro/roadmap/pull/1039) and help shape its future.
+
+## Sessions API
+
+The Astro Sessions API is now stable and ready for production!
+
+Sessions let you securely store data tied to a specific user as they navigate between pages on your site. Unlike cookies, sessions are stored on the server, so you can store larger amounts of data without worrying about size limits or security issues. They are useful for storing things like user data, shopping carts, and form state, and they work without any client-side JavaScript:
+
+
+```astro title="src/components/CartButton.astro" {2}
+---
+const cart = await Astro.session.get('cart');
+export const prerender = false;
+---
+
+<a href="/checkout">🛒 {cart?.length ?? 0} items</a>
+```
+
+They can be used in Astro components, pages, API endpoints, actions, and middleware. The API is the same for all of them:
+
+```ts title="src/actions/addToCart.ts" {9,10}
+import { defineAction } from "astro:actions";
+import { z } from "astro:schema";
+export const server = {
+  addToCart: defineAction({
+    input: z.object({
+      product: z.string(),
+    }),
+    async handler({ product }, { session }) {
+      const cart = await session.get('cart');
+      session.set('cart', [...cart, product]);
+    },
+  }),
+};
+```
```

Comment:
Adding some breathing space between imports and code to be consistent with other samples.
````suggestion
```ts title="src/actions/addToCart.ts" {9,10}
import { defineAction } from "astro:actions";
import { z } from "astro:schema";

export const server = {
  addToCart: defineAction({
    input: z.object({
      product: z.string(),
    }),
    async handler({ product }, { session }) {
      const cart = await session.get('cart');
      session.set('cart', [...cart, product]);
    },
  }),
};
```
````